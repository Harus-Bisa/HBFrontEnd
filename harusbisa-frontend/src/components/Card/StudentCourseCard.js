import React from "react";
import { Card, CardContent, CardActions, CardHeader} from "@material-ui/core";
import {connect} from "react-redux";
import "../../css/card.css";
import MenuOptions from "../MenuOptions/MenuOptions";

function mapStateToProps(state, currentProps){
    var courseId = currentProps.id 
    var courses = state.courses
    for (var i=0; i<courses.length; i++){
        if (courses[i].courseId === courseId){
            return {
                course: courses[i]
            }
        }
    }
    return null
}
function StudentCourseCard(props){
    return(
        <div className="col-md-6">
            <Card className="card student-card">
                <CardHeader
                    action={
                        <MenuOptions/>
                    }
                    title={<h5><a href={"/student/dashboard/"+props.course.courseId}>{props.course.courseName}</a></h5>}
                />
                <CardContent className="d-none d-md-block" style={{padding:'0px 16px 16px'}}>
                    <p>{props.course.startTerm} - {props.course.endTerm}</p>
                    <p>{props.course.instructors[0]}</p>
                </CardContent>
                <CardActions className="footer">
                    <p>{props.course.numberOfLectures} Sesi</p>
                </CardActions>
            </Card>
        </div>
    )
}

export default connect(mapStateToProps)(StudentCourseCard);