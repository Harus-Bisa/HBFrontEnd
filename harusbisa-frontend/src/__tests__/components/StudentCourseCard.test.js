import React from "react";
import StudentCourseCard from "../../components/Card/StudentCourseCard";
import {mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {Provider} from "react-redux";

const mockStore = configureMockStore();
describe("StudentCourseCard component", () =>{
    let initialState;
    let studentCourseCard;
    let store;
    beforeEach(() =>{
        initialState = {
            courses:[{
                "courseId": "5db492c349a67b33b8d0a2a2",
                "courseName": "Calculus I",
                "courseDescription": "The basic of calculus (derivatives).",
                "startTerm": "August 2019",
                "endTerm": "December 2019",
                "numberOfLectures": 5,
                "numberOfStudents": 21,
                "instructors": [
                  "John Doe",
                  "Bryan Smith"
                ],
                "joinCode": "123654"
              }]
        }
        store = mockStore(initialState);
        studentCourseCard = mount(
        <Provider store={store}>
            <StudentCourseCard id={initialState.courses[0].courseId}/>
        </Provider>)
    })

    it("Matches snapshot", () =>{
        expect(studentCourseCard.html()).toMatchSnapshot();
    })
})