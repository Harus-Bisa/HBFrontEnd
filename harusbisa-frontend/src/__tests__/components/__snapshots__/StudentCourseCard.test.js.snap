// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentCourseCard component Matches snapshot 1`] = `"<div class=\\"col-md-6\\"><div class=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root card student-card MuiPaper-rounded\\"><div class=\\"MuiCardHeader-root\\"><div class=\\"MuiCardHeader-content\\"><span class=\\"MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock\\"><h5><a href=\\"/student/dashboard/5db492c349a67b33b8d0a2a2\\">Calculus I</a></h5></span></div><div class=\\"MuiCardHeader-action\\"><div><button class=\\"MuiButtonBase-root MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div><div class=\\"MuiCardContent-root d-none d-md-block\\" style=\\"padding: 0px 16px 16px;\\"><p>August 2019 - December 2019</p><p>John Doe</p></div><div class=\\"MuiCardActions-root footer MuiCardActions-spacing\\"><p>5 Sesi</p></div></div></div>"`;
